poems_parser.py

```diff
@@ -1,67 +1,92 @@
-# Create file requirements.txt and put there all external modules (BeautifulSoup)
-# https://pip.readthedocs.io/en/1.1/requirements.html
-
-# Add .gitignore file to the repo, and configure it to ignore IDE specific folder ".idea"
-# https://git-scm.com/docs/gitignore
-
-# Please read how to create pull requests, it will be much more easier to review next changes
-# https://help.github.com/en/desktop/contributing-to-projects/creating-a-pull-request
 import urllib.request
 from bs4 import BeautifulSoup
 import os
 import argparse
-import codecs
+import re
+import operator
 
 
-# Open and read url
-# This is one line function and you use it only once in your code, it's not necessary.
+# Open resource
 def get_resource(url):
     resource = urllib.request.urlopen(url)
     return resource.read()
 
 
-# Parse html
-def parse_page(html):
+# parser
+def parse_page(html, url):
     res = []
     soup = BeautifulSoup(html, 'html.parser')
     table = soup.find('table', width="99%", border="0", cellspacing="2", cellpadding="2")
     for row in table.find_all('td', valign='top'):
-        res.append(row.find('a', class_='linki2'))
-    return res
+        poem_name = row.find('a', class_='linki2').text
+        poem_ur = url + row.find('a', class_='linki2')['href']
+        try:
+            poem_year = row.find('span').text
+        except AttributeError:
+            continue
+        poem_year = re.findall('[0-9]{4}', poem_year)
 
+        res.append([poem_name, poem_year[0], poem_ur])
```

Comment:
In case AttributeError you skip poems name, in this case it's better to handle year processing like this:
```suggestion
        try:
            poem_year = row.find('span').text
            poem_year = re.findall('[0-9]{4}', poem_year)
            poem_year = poem_year[0]
        except AttributeError:
            poem_year = ''
        res.append([poem_name, poem_year, poem_ur])
```